lib/types.ts

```diff
@@ -5,3 +5,16 @@
 import astTypes from "ast-types";
 export default astTypes;
 export * from "ast-types";
+
+// TODO Get these types from ast-types.
+
+export type Position = {
```

Comment:
In order to obtain these types from `ast-types`, we need to be a little more careful about making optional fields optional, rather than just giving them a default value (e.g. the `source` field of `SourceLocation`, which a non-optional `string | null` according to `ast-types`).